gui/integrity-dialog.cpp

```diff
@@ -146,6 +146,38 @@ IntegrityDialog::IntegrityDialog(Common::String endpoint, Common::String domain)
 IntegrityDialog::~IntegrityDialog() {
 }
 
+
+bool IntegrityDialog::progressUpdateCallback(int bytesProcessed) {
+	if(g_checksum_state->dialog->_close)
+		return false;
+	
+	g_checksum_state->calculatedSize += bytesProcessed;
+
+	if (g_system->getMillis() > g_checksum_state->lastUpdate + 500) {
+		g_checksum_state->lastUpdate = g_system->getMillis(); 
+		g_checksum_state->dialog->sendCommand(kDownloadProgressCmd, 0);
+	}
+
+	Common::Event event;
+	g_checksum_state->dialog->pollEvent(event);
+
+	return true;
+};
+
+
+void IntegrityDialog::pollEvent(Common::Event &event) {
+	if (g_system->getEventManager()->pollEvent(event)) {
+		static uint32 lastEventPoll = 0;
```

Comment:
This is very bad design. Avoid static variabless in class methods. Move this variable to the class.